packages/api/src/auth/decoders/clerk.ts

```diff
@@ -0,0 +1,45 @@
+import { sessions, users } from '@clerk/clerk-sdk-node'
+import type { APIGatewayProxyEvent, Context as LambdaContext } from 'aws-lambda'
+
+import type { GlobalContext } from 'src/globalContext'
+
+export type Req = {
+  event: APIGatewayProxyEvent
+  context: GlobalContext & LambdaContext
+}
+
+export const clerk = async (token: string, req: Req) => {
+  if (!process.env.CLERK_API_KEY) {
+    console.error('CLERK_API_KEY env var is not set.')
+    throw new Error('CLERK_API_KEY env var is not set.')
+  }
+
+  try {
+    // Clerk sessions are a combination of a clerk "current session id", which we store
+    // in the Redwood auth token, and the __session cookie, which contains a second session
+    // bearer token. The two tokens together define which device is browsing and as who.
+    const clerkCookieName = '__session'
+    const cookies = req.event.headers['cookie']?.split(';').map((c) => c.trim())
+    const sessionCookie = cookies
+      ?.find((c) => c.startsWith(clerkCookieName + '='))
+      ?.substring(clerkCookieName.length + 1)
+
+    if (!sessionCookie || sessionCookie.length < 1) {
+      return Promise.reject(new Error('Clerk __session token is not set'))
+    }
+
+    const session = await sessions.verifySession(token, sessionCookie)
+    if (!session.userId) {
+      return Promise.reject(new Error('Session invalid'))
+    }
+
+    const user = await users.getUser(session.userId)
+
+    return {
+      ...user,
+      roles: user.publicMetadata['roles'] ?? [],
+    }
+  } catch (error) {
+    return Promise.reject(error)
```

Comment:
What do you think of remove the try catch since we're not doing anything with this error?

Reply:
👍  sure - was just modeling it over-strictly on the other decoders but agree with the simpler catch it doesn't make sense anymore.